core/src/main/java/org/jdbi/v3/core/qualifier/Qualifiers.java

```diff
@@ -64,17 +57,16 @@ public void setRegistry(ConfigRegistry registry) {
      * which searches for annotations themselves annotated with {@link Qualifier}.
      *
      * Resolvers should return static results that don't vary over time; therefore their results are cached.
-     * Adding a resolver will clear the local qualifier resolution cache, but is currently not guaranteed to have any effect on {@code SqlObjects}.
-     * Add any resolvers you may need <em>before</em> using any qualifiers-related features!
+     * Add any resolvers you may need <em>before</em> using any types that may be qualified - cached qualifiers
+     * are not guaranteed to update!
      *
      * @param resolver the resolver to be included in qualifier resolution
      * @return this
      */
-    public Qualifiers addResolver(Function<AnnotatedElement, Set<Annotation>> resolver) {
-        resolvers.add(resolver);
-        if (registry != null) {
-            QUALIFIER_CACHE.clear(registry);
-        }
+    @SafeVarargs
+    public final Qualifiers registerQualifier(Class<? extends Annotation>... annotations) {
+        registered.addAll(Arrays.asList(annotations));
+        QUALIFIER_CACHE.clear(registry);
```

Comment:
A local config change (e.g. on a SQL Object method) invalidates the global qualifier cache? This means that qualifiers could get loaded and reloaded and get different values, depending on the last time a SQL object registered a qualifier.

It's even worse if you have two different SQL objects registering different qualifiers.

If the qualifier cache is global, then the qualifier registry needs to be global too.